### PR TITLE
merge sort added more efficient

### PR DESCRIPTION
**Pull request by BamaCharanChhandogi**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by BamaCharanChhandogi** [comment]
hey! @pankajc007 could you check?

File: sorting/merge_sort.c [hunk 1]
```diff
@@ -1,139 +1,118 @@
 /**
- * @file
- * @brief Implementation of [merge
```
Comment (realstealthninja) [inline sorting/merge_sort.c:3]: please restore these lines

File: sorting/merge_sort.c [hunk 2]
```diff
@@ -1,139 +1,118 @@
...
- * sort](https://en.wikipedia.org/wiki/Merge_sort) algorithm
+ *@author [Bama Charan Chhandogi](https://github.com/BamaCharanChhandogi)
  */
+#include <assert.h> // for Testing
 #include <stdio.h>
 #include <stdlib.h>
```
Comment (realstealthninja) [inline sorting/merge_sort.c:6]:
```suggestion
#include <assert.h> /// For assert
#include <stdio.h> /// For I/O operations
#include <stdlib.h> /// For memory operations
```

File: sorting/merge_sort.c [hunk 3]
```diff
@@ -1,139 +1,118 @@
...
-/**
- * @addtogroup sorting Sorting algorithms
- * @{
- */
-/** Swap two integer variables
- * @param [in,out] a pointer to first variable
- * @param [in,out] b pointer to second variable
- */
-void swap(int *a, int *b)
+// Function to merge two sorted subarrays into one sorted array
```
Comment (realstealthninja) [inline sorting/merge_sort.c:7]: please follow proper documentation guidelines

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: sorting/merge_sort.c [hunk 4]
```diff
@@ -2,138 +2,121 @@
...
-        printf("Can't Malloc! Please try again.");
-        exit(EXIT_FAILURE);
-    }
-    int c = l;
-    int p1, p2;
-    p1 = l;
-    p2 = ((l + r) / 2) + 1;
-    while ((p1 < ((l + r) / 2) + 1) && (p2 < r + 1))
+    // Create temporary arrays to store the left and right subarrays
+    int L[n1], R[n2];
```
Comment (vil02) [inline sorting/merge_sort.c:19]: `ISO C++ forbids variable length array` - I think you should use `malloc` here (and of course `free` later).

**Pull request closed without merging**